### PR TITLE
API Journalist can create article

### PR DESCRIPTION
**Pull request by JaimeCrz**
[PT Story](https://www.pivotaltracker.com/story/show/171811192)

```
As a journalist,
In order to write articles
I would like to be able to receive an article from admin front-end and save it to database
```
## Changes proposed in this pull request:

* Added create Article function in controller and request spec for it
* Fixed new POST Routes

## What I have learned working on this feature:
* Learn how to use new_record? and persisted? 
* We refreshed knowledge in configuring permitted params.

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
    end
   end
 
   def show
-    article = Article.find(params[:id]) rescue nil
+    article = begin
+                Article.find(params[:id])
+              rescue StandardError
+                nil
+              end
```
Comment (tochman) [inline app/controllers/api/articles_controller.rb]: I see no reason for a begin rescue block here. Also, is this show action in scope of this story?

File: app/controllers/api/articles_controller.rb [hunk 2]
```diff
@@ -1,19 +1,43 @@
...
     if article.nil?
-      render json: { message: "Sorry your article wasn't found" }, status: 404 
+      render json: { message: "Sorry your article wasn't found" }, status: 404
     else
       render json: article, status: 200
     end
   end
+
+  def create
+    article = Article.new(article_params)
```
Comment (tochman) [inline app/controllers/api/articles_controller.rb]: You can (and should) avoid nested `if` statements.

How about calling `Article.create` and use `article.persisted?` in the `if` statement? Then you can skip the `.new_record?` check.

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   namespace :api do
     resources :articles, only: %i[index show], constraints: { format: 'json' }
+    post 'articles/create'
```
Comment (tochman) [inline config/routes.rb]: please remove this and add `:create` to the resources helper above

File: spec/requests/api/journalist_can_create_article_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+RSpec.describe Api::ArticlesController, type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  describe 'POST /api/articles_controller' do
+    before do
+      post '/api/articles/create',
+           params: {
+             title: 'Coronavirus',
+             teaser: 'Things are bad',
+             content: 'It will get worse.'
```
Comment (tochman) [inline spec/requests/api/journalist_can_create_article_spec.rb:13]: IMO, the conventional way to send this off is wrapped in an `'article'` key. It will be parsed by your controller in that way and require you to sanitize params differently than you do at the moment.

File: spec/requests/api/user_can_see_all_articles_spec.rb [hunk 5]
```diff
@@ -9,14 +11,14 @@
...
+      get '/api/articles'
+    end
 
     it 'No articles been found' do
       expect(response.status).to eq 404
     end
   end
-end
\ No newline at end of file
+end
```
Comment (tochman) [inline spec/requests/api/user_can_see_all_articles_spec.rb:24]: There are no changes here. Now, if you format your code properly to start with, you don't have to see these files as part of your PR's

File: config/routes.rb (hunk 3 above)
Comment (JaimeCrz) [reply config/routes.rb]: done!

File: spec/requests/api/user_can_see_all_articles_spec.rb (hunk 5 above)
Comment (JaimeCrz) [reply spec/requests/api/user_can_see_all_articles_spec.rb:24]: We will keep this in mind! Sorry, we started to use solargraph recently in specs files, as it was unable by default.

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (JaimeCrz) [reply app/controllers/api/articles_controller.rb]: fixed this issue

**Comment by johanbounce** [comment]
Looks fine by me!
